minecraft-plugin/src/main/java/com/pvchat/proximityvoicechat/plugin/PlayerVolumeServer.java

```diff
@@ -0,0 +1,117 @@
+package com.pvchat.proximityvoicechat.plugin;
+
+import com.anthonynsimon.url.URL;
+import com.anthonynsimon.url.exceptions.MalformedURLException;
+import com.cedarsoftware.util.io.JsonWriter;
+import com.pvchat.proximityvoicechat.plugin.distanceMatrix.PlayerVolumeData;
+import org.bukkit.Bukkit;
+import org.java_websocket.WebSocket;
+import org.java_websocket.handshake.ClientHandshake;
+import org.java_websocket.server.WebSocketServer;
+
+import java.net.InetSocketAddress;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class PlayerVolumeServer extends WebSocketServer {
+
+    private final Map<DiscordUserID, WebSocket> openConnections;
+    private final Logger logger;
+    private final String DISCORD_ID_QUERY_KEY = "discordID";
+    private final ProximityVoiceChat pluginInstance;
+
+    public PlayerVolumeServer(int webSocketPort, ProximityVoiceChat pluginInstance) {
+        super(new InetSocketAddress(webSocketPort));
+        this.pluginInstance = pluginInstance;
+        openConnections = new ConcurrentHashMap<>();
+        logger = Bukkit.getLogger();
+    }
+
+    public Consumer<List<PlayerVolumeData>> sendPlayerVolumeMatrix = new Consumer<>() {
+        @Override
+        public void accept(List<PlayerVolumeData> matrixData) {
+
+            if (matrixData == null) return;
```

Comment:
Use `Optional` if you want a parameter to be optional. Do not use `null` as it is very error prone.